easybuild/easyblocks/generic/bundle.py

```diff
@@ -324,6 +323,38 @@ def install_step(self):
             # close log for this component
             comp.close_log()
 
+    def make_module_req_guess(self):
+        """
+        Set module requirements from all comppnents, e.g. $PATH, etc.
+        During the install step, we only set these requirements temporarily.
+        Later on when building the module, those paths are not considered.
+        Therefore, iterate through all the components again and gather
+        the requirements.
+
+        Do not remove duplicates or check for existance of folders,
+        as this is done in the generic EasyBlock while creating
+        the modulefile already.
+        """
+        # Start with the paths from the generic EasyBlock.
+        # If not added here, they might be missing entirely and fail sanity checks.
+        final_reqs = super(Bundle, self).make_module_req_guess()
+
+        for cfg, comp in self.comp_instances:
+            self.log.info("Gathering module paths for component %s v%s", cfg['name'], cfg['version'])
+            reqs = comp.make_module_req_guess()
```

Comment:
What if one of the component itself is a `Bundle` instance, that may result in an infinite loop I think?

Reply:
Hmm... a `Bundle` instance should gather the requirements of the components. If one of the components is also a `Bundle`, we should collect all of them recursively.

An infinite loop can be achieved if one EasyBlock tries to call this function via `super`. Can we prevent that?

Reply:
@boegel I don't see why there would be an infinite loop here, the list of components is finite and equal to whatever the easyconfig defines. Even if there is a bundle in the component list, it is also defined in the easyconfig.
This is not like looping over dependencies, with multiple layers of depth.

Reply:
Having a `Bundle` as component of another `Bundle` is explicitly forbidden. See https://github.com/easybuilders/easybuild-easyblocks/blob/28b208493ca35bbd73917c047356d3b92c51db7c/easybuild/easyblocks/generic/bundle.py#L142
So this is not the cause of the issue (unless some unit test does some weird stuff and hijacks that check in `Bundle`)

Reply:
Apparently there is a general bug in this check, as it does not cover easyblocks based on `Bundle`, like `PythonBundle` or `OpenSSL_wrapper`. So that is one thing that can improve.
However, it is already impossible to make a nested bundle in practice because the resulting construct always errors out. If the _bundle_ component has sources, it results in error about _"bundles cannot have sources"_; and if it does not have `sources` it results in error about _"missing sources"_.